ztp/tools/pgt2acm/main.go

```diff
@@ -0,0 +1,359 @@
+	err = convertAllPGTFiles(preRenderPatchKindList, allFilesInInputPath, inputFile, outputDir, schema, workaroundPlacement)
+	if err != nil {
+		fmt.Printf("Could not convert PGT files, err: %s", err)
+		os.Exit(1)
+	}
+
+	if NSYAML != nil && *NSYAML != "" {
+		err = fileutils.CopyAndProcessNSAndKustomizationYAML(*NSYAML, *inputFile, *outputDir, *skipDefaultPlacementBindings)
+		if err != nil {
+			fmt.Printf("Could not post-process %s and %s files, err: %s", *NSYAML, fileutils.KustomizationFileName, err)
+		}
+	}
+
+	if *generateACMPolicies {
+		err = renderpolicies.RenderAndWriteTemplateToYAML(*outputDir, renderpolicies.AcmGenRenderedYAMLFileName)
+		if err != nil {
+			fmt.Printf("Could not generate ACMGen policies, err: %s", err)
+			os.Exit(1)
+		}
+
+		err = renderpolicies.RenderAndWriteTemplateToYAML(*inputFile, renderpolicies.PgtRenderedYAMLFileName)
+		if err != nil {
+			fmt.Printf("Could generate PGT policies, err: %s", err)
+			os.Exit(1)
+		}
+	}
+}
+
+func convertAllPGTFiles(preRenderPatchKindList, allFilesInInputPath []string, inputFile, outputDir, schema *string, workaroundPlacement *bool) (err error) {
+	for _, file := range allFilesInInputPath {
+		var kindType fileutils.KindType
+		kindType, err = fileutils.GetManifestKind(file)
+		if err != nil {
+			return fmt.Errorf("could not get manifest kind for file:%s, err: %s", file, err)
+		}
+		if kindType.Kind != "PolicyGenTemplate" {
+			continue
+		}
+		// Get the relative path
+		var relativePath string
+		relativePath, err = filepath.Rel(*inputFile, file)
+		if err != nil {
+			return fmt.Errorf("error getting relative path, err:%s", err)
+		}
+		err = convertPGTtoACM(*outputDir, file, filepath.Join(*outputDir, fileutils.PrefixLastPathComponent(relativePath, fileutils.ACMPrefix)), *schema, preRenderPatchKindList, workaroundPlacement)
+		if err != nil {
+			return fmt.Errorf("failed to convert PGT to ACMGen, err=%s", err)
+		}
+	}
+	return nil
+}
+
+// Converts an PGT file to a ACM Gen Template file
+func convertPGTtoACM(outputDir, inputFile, outputFile, schema string, preRenderPatchKindList []string, workaroundPlacement *bool) (err error) {
+	policyGenFileContent, err := os.ReadFile(inputFile)
+	if err != nil {
+		return fmt.Errorf("unable to open file: %s, err: %s ", inputFile, err)
+	}
+	policyGenTemp := pgtformat.PolicyGenTemplate{}
+
+	err = yaml.Unmarshal(policyGenFileContent, &policyGenTemp)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal PolicyGenTemplate data from %s: %s", inputFile, err)
+	}
+
+	rootName := policyGenTemp.Metadata.Name
+	acmGenTempConversion := acmformat.AcmGenTemplate{}
+
+	seenPoliciesMap := map[string]bool{}
+	for srcFileIndex := range policyGenTemp.Spec.SourceFiles {
+		seenPoliciesMap[policyGenTemp.Spec.SourceFiles[srcFileIndex].PolicyName] = true
+	}
+
+	var seenPoliciesSorted []string
+	for policyName := range seenPoliciesMap {
+		seenPoliciesSorted = append(seenPoliciesSorted, policyName)
+	}
+
+	sort.Strings(seenPoliciesSorted)
```

Comment:
Why do we need to sort policy names?

Reply:
Yes, this is used later to consistently compare side by side rendered ACM policies from PGT and ACM PolicyGen